### PR TITLE
Libcloud 574 routes and route tables

### PR DESCRIPTION
**Pull request by zerthimon**
Support for EC2 Route Tables and Routes.

### CODE REVIEW

**Comment by zerthimon** [comment]
@Kami is there anything else I should do here ?

File: libcloud/compute/drivers/ec2.py [hunk 1]
```diff
@@ -1730,6 +1739,105 @@ def __repr__(self):
...
 
+class EC2RouteTable(object):
+    """
+    Class which stores information about VPC Route Tables.
+
+    :param      id: The ID of the route table.
+    :type       id: ``str``
+
+    :param      routes: A list of routes in the route table.
+    :type       routes: ``list`` of :class:`EC2Route`
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: Please add `.` before the classes which are located in the same module. In this case:

``` python
 :type       routes: ``list`` of :class:`.EC2Route`
```

File: libcloud/compute/drivers/ec2.py [hunk 2]
```diff
@@ -1730,6 +1739,105 @@ def __repr__(self):
...
+    def __init__(self, id, routes, subnet_associations,
+                 propagating_gateway_ids, extra=None):
+        self.id = id
+        self.routes = routes
+        self.subnet_associations = subnet_associations
+        self.propagating_gateway_ids = propagating_gateway_ids
+        self.extra = extra or {}
+
+    def __repr__(self):
+        return (('<EC2RouteTable: id=%s') % (self.id))
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: Missing closing >

File: libcloud/compute/drivers/ec2.py [hunk 3]
```diff
@@ -1730,6 +1739,105 @@ def __repr__(self):
         return (('<VPCInternetGateway: id=%s>') % (self.id))
 
 
+class EC2RouteTable(object):
+    """
+    Class which stores information about VPC Route Tables.
+
+    :param      id: The ID of the route table.
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: Please move docstring which documents the arguments inside the `__init__` method.

Same goes for other classes.

File: libcloud/compute/drivers/ec2.py [hunk 4]
```diff
@@ -1730,6 +1739,105 @@ def __repr__(self):
...
+        self.gateway_id = gateway_id
+        self.instance_id = instance_id
+        self.owner_id = owner_id
+        self.interface_id = interface_id
+        self.state = state
+        self.origin = origin
+        self.vpc_peering_connection_id = vpc_peering_connection_id
+
+    def __repr__(self):
+        return (('<EC2Route: cidr=%s') % (self.cidr))
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: Same here, missing >

File: libcloud/compute/drivers/ec2.py [hunk 5]
```diff
@@ -3927,6 +4035,311 @@ def ex_detach_internet_gateway(self, gateway, network):
 
         return element == 'true'
 
+    def ex_list_route_tables(self, rtable_ids=None, filters=None):
```
Comment (Kami) [inline libcloud/compute/drivers/ec2.py]: Probably would be better to be explicit and and consistent with the method name and call the argument `route_table_ids`

**Comment by Kami** [comment]
Ok, reviewed the whole pull request.

There are some minor issues left, but it mostly looks good. Next step would be adding tests for this new functionality. Let us know if you need help with that.

**Comment by Kami** [comment]
@zerthimon Let me know once those issues have been addressed and I will get it merged into trunk.

**Comment by zerthimon** [comment]
@Kami 

Fixed issues that you found during code review.
Also changed ex_create_route() and ex_replace_route() to accept objects and extract IDs, instead of passing IDs as strings.

**Comment by Kami** [comment]
Merged into trunk. Thanks.

**Pull request closed without merging**